### PR TITLE
fix: classify toa attempts as safe deaths

### LINKED ISSUE #316: Death notifs: TOA safe deaths
Currently, all TOA deaths trigger the discord notification, even the safe deaths with the "Ignore Safe Deaths" setting toggled on.
My suggestion is to have this only cause the event when you've run out of lives. If the issue concerns how to know which one is safe, have the screenshot taken on death but wait for the "You failed to survive the Tombs of Amascut." message before actually sending the notification.

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #316

### CODE REVIEW

File: src/main/java/dinkplugin/util/WorldUtils.java [hunk 1]
```diff
@@ -105,6 +106,12 @@ public boolean isPlayerOwnedHouse(int regionId) {
     public boolean isSafeArea(Client client) {
         int regionId = getLocation(client).getRegionID();
 
+        if (isAmascutTombs(regionId)) {
+            // ToA is technically a dangerous activity, but multiple attempts can be permitted
+            // the real TOA death is detected via game message in death notifier
+            return true;
+        }
```
Comment (pajlada) [inline src/main/java/dinkplugin/util/WorldUtils.java:113]: Currently this makes ToA safe for HCIM, I have a feeling that's not how it works

**Comment by zneix** [comment]
Are all ToA deaths meant to be safe though? Like pajlada said, you do lose HCIM status if you die inside the tombs at all - you might not even have a full wipe / lose a team life, but if you die yourself the status is gone - much like in ToB.
Even though, regardless of this I'm not sure if I like this change at all. It is technically safe and kinda like deaths in CoX but I just can't pinpoint what's different about it.

File: src/main/java/dinkplugin/util/WorldUtils.java (hunk 1 above)
Comment (zneix) [reply src/main/java/dinkplugin/util/WorldUtils.java:113]: > https://youtu.be/GzwCCghNckw?t=36 had the death message, not sure what invocations were used so this should be ok

FWIW, this looks like Hardcore Run invocation where you get only 1 team life, however he was also alone in the raid, so his 1 death was considered a wipe (and full team wipes result in a "raid life" loss), and also 1 wipe (hardcore run) is considered a failed raid, so you're thrown out.

**Comment by pajlada** [comment]
To clarify: ToA is *not* a safe activity for Solo Hardcore Ironmen or Group Hardcore Ironmen - one death means loss of their Hardcore status or loss of one of their groups' lives.
I think we shouldn't rely on the messaging for those players, but rather just check the players status to figure out whether the death is safe or not.
Thanks @zneix !

**Comment by pajlada** [comment]
The downside to this approach is that the notification won't fire until you're gone from the place you died

Screenshot with "Ignore safe deaths" disabled:
image

Screenshot with "Ignore safe deaths" enabled, but losing my last life:
image

How janky is it to take a screenshot on death, and only fire it if it turns out it was the last life?

**Review by pajlada — APPROVED** [review]

**Review by Felanbird — APPROVED** [review]
this seems fine 👍 - we'll probably get an issue about this change but changing it seems correct

**Pull request merged**